examples/ensemble/plot_gradient_boosting_categorical.py

```diff
@@ -114,11 +125,11 @@
 
 # The ordinal encoder will first output the categorical features, and then the
 # continuous (passed-through) features
-categorical_mask = [True] * n_categorical_features + [False] * n_numerical_features
+categorical_mask = [True] * k_columns + [False] * k_columns
```

Comment:
This code is not robust. It will fail if a user change `k_columns` to a value larger than 34, because the number of features will be smaller than `2 * k_columns`. We should use `n_categorical_features` and `n_numerical_features`, computed after the reduction to at most `k_columns` features.